sdk/database/dbplugin/v5/testing/test_helpers.go

```diff
@@ -22,19 +22,47 @@ func getRequestTimeout(t *testing.T) time.Duration {
 	return dur
 }
 
-func AssertInitialize(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
+// AssertInitializeCircleCiTest help to diagnose CircleCI failures within AssertInitialize for mssql tests failing
+// with "Failed to initialize: error verifying connection ...". This will now mark a test as failed instead of being fatal
+func AssertInitializeCircleCiTest(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
 	t.Helper()
+	maxAttempts := 5
+	var resp dbplugin.InitializeResponse
+	var err error
+
+	for i := 1; i <= maxAttempts; i++ {
+		resp, err = verifyInitialize(t, db, req)
+		if err != nil {
+			t.Errorf("Failed AssertInitialize attempt: %d with error:\n%+v\n", i, err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		if i > 1 {
+			t.Logf("AssertInitialize worked the %d time around with a 1 second sleep", i)
```

Comment:
```suggestion
			t.Errorf("AssertInitialize worked the %d time around with a 1 second sleep", i)
```

?

this is what we discussed in slack i believe

Reply:
might also be good to comment that we want the test to fail if it doesn't work the first time?

Reply:
ohhhh oops missed the `Errorf` above!